packages/number-field/src/NumberField.ts

```diff
@@ -214,7 +228,13 @@ export class NumberField extends TextfieldBase {
     private valueBeforeFocus: string = '';
     private isIntentDecimal: boolean = false;
 
-    private convertValueToNumber(value: string): number {
+    private convertValueToNumber(inputValue: string): number {
+        // Normalize full-width characters to their ASCII equivalents
```

Comment:
Without normalizing this value (remapping the multi-byte characters)  before the input manipulation, we would get `NaN`.